### ISSUE TITLE
Simple authentication doesn't work with MQTT 5

### ISSUE THREAD

**Issue by anaganisk** [vernemq/vernemq#1826]
### Environment

- VerneMQ Version: 1.11.0-alpine (latest available for docker)
- OS: Kubernetes / Linux
- Erlang/OTP version (if building from source): N/A
- VerneMQ configuration (vernemq.conf) or the changes from the default - Default
- Cluster size/standalone: 1

### Expected behaviour
Simple authentication using username and password must work using mqtt 5

### Actual behaviour
I used MQTT version 5 to connect using username and password but when I use vmq-admin session show the --user column shows undefined, and it is affecting ACLs.
If I switch to Mqtt 3 it works.
I tested using nodejs mqtt client also java and python client to rule out client issues, all have the same result.

to reproduce

```js
var mqtt = require("mqtt");


require('net').createServer().listen().addListener("listening", () => {
    var client = mqtt.connect("xxxxxxx", {
        protocolVersion: 5,
        username: "xxxxx",
        password: "xxxxx",
        clientId: "TESTCLIENTWEB"
      });
      
      client.on("connect", function () {
        client.subscribe("#", function (err) {
          if (!err) {
            client.publish("presence", "Hello mqtts");
            client.publish("presence", "Hello mqtts");
          }
        });
      });
      
      client.on("message", function (topic, message) {
        // message is Buffer
        console.log(message.toString());
      });
});

```
Execute in node
```bash
npm init
npm install mqtt
node index.js
```

**Comment by ioolkos** [comment]
@anaganisk Thanks, you either  found a pretty big bug or regression here, or I don't yet have the full picture.
But I can reproduce this, it seems:

```sh
➜  ~ mosquitto_sub -d -h 127.0.0.1 -p 1887 -u bla -P blu -t testing -i me -V 5   

Client me sending CONNECT
Client me received CONNACK (0)
Client me sending SUBSCRIBE (Mid: 1, Topic: testing, QoS: 0, Options: 0x00)
Client me received SUBACK
Subscribed (mid: 1): 0
```
Get the session Pid: (we already see username is 'undefined')
```sh
➜  ~ sudo vmq-admin session show --client_id --user --session_pid
+-----------+-------------+-----------+
| client_id | session_pid | user      |
+-----------+-------------+-----------+
| me        | <0.745.0>   | undefined |
+-----------+-------------+-----------+
```
Attach to Verne and get the session state, to make sure it's not just the query in `session show`: (that second 'undefined' should be the username)
```erlang
(VerneMQ@127.0.0.1)8> rp(recon:get_state(<0.745.0>)).
{st,#Port<0.40>,<<>>,vmq_mqtt5_fsm,
    {connected,{state,1,
                      {[],<<"me">>},
                      undefined,#{},[],
                      {{127,0,0,1},51092},
                      undefined,60,undefined,true,0,5,<0.748.0>,
                      {1623,912405,441207},
                      {1623,911985,4468},
                      undefined,true,100,prefer_local,0,false,vmq_reg_trie,
                      {cap_settings,false,false,false,false},
                      0,0,#{},#{},20,65535,0,0,#{},undefined}},
    {tcp,tcp_closed,tcp_error},
    [],false,<0.436.0>}
ok

```

@anaganisk questions for you: you do authenticate username/password, right? or is it just happening with allow_anonymous? also, do you happen to have tested this for older releases?

---
https://github.com/sponsors/vernemq

**Comment by anaganisk** [comment]
@ioolkos 
Use im trying auth with username and password
allow_anonymous is set to on, we need few acls that allow for anonymous users and few for authenticated users, also we're using file base authentication,
as expected all the anonymous connections show up as undefined.
But when I connect with username and password, username shows up only for MQTT 3 or 4 connections, MQTT5 just shows undefined.

I tested it on 1.10.1, didnt work then upgraded to 1.11.0, didn't work here either.

**Comment by ioolkos** [comment]
@anaganisk okay, thanks for confirming.

---
https://github.com/sponsors/vernemq

**Comment by ioolkos** [comment]
@anaganisk Ok, so then there's a difference in how we implemented anonymous case for v4 and v5. This makes the bug much less critical (authenticated clients should have correct username, also with v5).

What we probably should do is have a consistent approach. MQTT v5 auth is a little bit different, so I'll have to check whether this is possible and makes sense.

Just to be clear (also for v4): `allow_anonymous` will completely shortcut authentication (not authorization). It will *not* check whether the provided username/pw is correct.

**Comment by anaganisk** [comment]
@ioolkos seems like you're right. If I set allow_anonymous=off and use username password authentication it works.
May be its worthy of a mention on the file based authentication docs since it may paint a wrong picture due to ACL example, until the issue is may be fixed.

**Comment by ioolkos** [comment]
@anaganisk see PR #1870 

---
:point_right: Thank you for supporting VerneMQ: https://github.com/sponsors/vernemq
:point_right: Using the binary VerneMQ packages (.deb/.rpm/Docker) requires a subscription.

**Comment by ioolkos** [comment]
Fixed in 1.12.2
